FRENCH/src/ch17-02-trait-objects.md

```diff
@@ -0,0 +1,1020 @@
+pointe à la fois vers une instance d'un type implémentant le trait indiqué ainsi
+que vers une table utilisée pour chercher les méthodes de trait de ce type à
+l'exécution. Nous créons un objet trait en indiquant une sorte de pointeur, tel
+qu'une référence `&` ou un pointeur intelligent `Box<T>`, puis le mot-clé `dyn`
+et enfin le trait en question. (Nous expliquerons pourquoi les objets traits
+doivent utiliser un pointeur dans [une section][dynamically-sized]<!-- ignore -->
+du chapitre 19.) Nous pouvons
+utiliser des objets traits à la place d'un type générique ou concret. Partout où
+nous utilisons un objet trait, le système de types de Rust s'assurera à la
+compilation que n'importe quelle valeur utilisée dans ce contexte implémentera
+le trait de l'objet trait. Ainsi, il n'est pas nécessaire de connaître tous les
+types possibles à la compilation.
+
+<!--
+We’ve mentioned that in Rust, we refrain from calling structs and enums
+“objects” to distinguish them from other languages’ objects. In a struct or
+enum, the data in the struct fields and the behavior in `impl` blocks are
+separated, whereas in other languages, the data and behavior combined into one
+concept is often labeled an object. However, trait objects *are* more like
+objects in other languages in the sense that they combine data and behavior.
+But trait objects differ from traditional objects in that we can’t add data to
+a trait object. Trait objects aren’t as generally useful as objects in other
+languages: their specific purpose is to allow abstraction across common
+behavior.
+-->
+
+Nous avons mentionné qu'en Rust, nous nous abstenons de qualifier les structures
+et énumérations d'*objets* pour les distinguer des objets des autres langages.
+Dans une structure ou une énumération, les données dans les champs de la
+structure et le comportement dans les blocs `impl` sont séparés, alors que dans
+d'autres langages, les données et le comportement se combinent en un concept
+souvent qualifié d'objet. En revanche, les objets traits ressemblent davantage
+aux objets des autres langages dans le sens où ils combinent des données et du
+comportement. Mais les objets traits diffèrent des objets traditionnels dans le
+sens où on ne peut pas ajouter des données à un objet trait. Les objets traits
+ne sont généralement pas aussi utiles que les objets des autres langages : leur
+but spécifique est de permettre de construire des abstractions de comportements
+communs.
+
+<!--
+Listing 17-3 shows how to define a trait named `Draw` with one method named
+`draw`:
+-->
+
+L'encart 17-3 illustre la façon de définir un trait nommé `Affichable` avec une
+méthode nommée `afficher` :
+
+<!--
+<span class="filename">Filename: src/lib.rs</span>
+-->
+
+<span class="filename">Fichier : src/lib.rs</span>
+
+<!--
+```rust
+pub trait Draw {
+    fn draw(&self);
+}
+```
+-->
+
+```rust
+pub trait Affichable {
+    fn afficher(&self);
+}
+```
```

Comment:
I can see two identical code blocks here: https://jimskapt.github.io/rust-book-fr/ch17-02-trait-objects.html#d%C3%A9finir-un-trait-pour-du-comportement-commun
But i don't know where he is in the code xD